src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -565,6 +565,17 @@ public List<UUID> getAccessibleSnapshots(List<RasDbgapPermissions> permissions)
     return accessibleSnapshots;
   }
 
+  /**
+   * @return a list of all snapshot IDs, including those exclusively locked
```

Comment:
RE: listing locked snapshots: I wonder if the end-users plan on turning around and immediately running retrieveSnapshot on the ids. As you mentioned in your spike doc, this makes me want to prioritize the [DR-1907](https://broadworkbench.atlassian.net/browse/DR-1907) work.

Reply:
This is a good point!

@nmalfroy, do you know whether HCA facilitates individual snapshot retrieval off of the enumeration, either programmatically or on demand?

I wouldn't be opposed to altering this method to stay consistent with existing snapshot enumeration behavior (exclude exclusively locked snapshot), then handle it when hopefully working on DR-1907.  I see that it's an older ticket and I fear it's been stuck in JIRA purgatory, haha.

Reply:
We talked about this in stand-up today.  A snapshot should only be exclusively locked while it's being created or deleted so we don't think this inconsistency will pose a problem for HCA.